src/Compilers/CSharp/Test/Semantic/Semantics/TargetTypedDefaultTests.cs

```diff
@@ -322,6 +322,38 @@ static void M()
         }
 
         [Fact]
+        public void BadUnaryOperator()
+        {
+            string source = @"
+class C<T>
+{
+    static void M()
+    {
+        var a = +default;
+        var b = -default;
+        var c = ~default;
+        var d = !default;
+    }
+}
+";
+            var comp = CreateStandardCompilation(source, parseOptions: TestOptions.Regular7_1);
+            comp.VerifyDiagnostics(
+                // (6,17): error CS0023: Operator '+' cannot be applied to operand of type 'default'
```

Comment:
I agree the message is not the best.
This also appear when you type `default is long`:
```C#
    // (6,30): error CS0023: Operator 'is' cannot be applied to operand of type 'default'
                //         System.Console.Write(default is long);
                Diagnostic(ErrorCode.ERR_BadUnaryOp, "default is long").WithArguments("is", "default").WithLocation(6, 30),
```